### PR TITLE
Refactor docker compose services to expose ports instead of using host network

### PR DESCRIPTION
**Pull request by breadchris**
Up until now, the lunasec stack in the docker compose file was using the host network to be able to communicate with each service. This was not clean and could lead to a lot of head scratching scenarios during development of what services exposed what ports, a service binding to unexpected port numbers on the host, or colliding with other services running on the host machine. Ports are now explicitly set in the service to be bound to the host network.

The main challenge with this change is to make sure all configuration values that reference hostnames/localhost are set appropriately for the right environments.

### CODE REVIEW

File: js/demo-apps/packages/react-front-end/cypress/plugins/index.js [hunk 1]
```diff
@@ -36,7 +36,39 @@ module.exports = (on, config) => {
...
+        printLogsToConsole: 'always'
     }
     require('cypress-terminal-report/src/installLogsPrinter')(on, options);
-}
+
+    on('before:browser:launch', (browser = {}, launchOptions) => {
+        // if (browser.family === 'chromium' && browser.name !== 'electron') {
+            // auto open devtools
+        launchOptions.args.push('--auto-open-devtools-for-tabs');
+        launchOptions.args.push('--unsafely-treat-insecure-origin-as-secure=http://tokenizer-backend:37766,http://application-back-end:3001,http://application-back-end:3002');
```
Comment (factoidforrest) [inline js/demo-apps/packages/react-front-end/cypress/plugins/index.js:47]: oh boy haha you went deep

File: js/demo-apps/packages/react-front-end/package.json [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-  "version": "1.0.5",
+  "version": "1.0.6-alpha.1",
   "license": "Apache-2.0",
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "test:e2e": "cypress run --browser chrome",
-    "cypress:open": "cypress open --browser chrome",
+    "test:url": "if [ -z ${APPLICATION_FRONT_END+x} ]; then echo 'http://localhost:3000'; else echo $APPLICATION_FRONT_END; fi",
```
Comment (factoidforrest) [inline js/demo-apps/packages/react-front-end/package.json:10]: in the future maybe we could run this kind of logic into our application code/test config.  I believe the cypress.js file can control this kind of thing programmatically..could be wrong though, not the end of the world either way

File: js/sdks/packages/cli/src/docker-compose/lunasec-stack.ts [hunk 3]
```diff
@@ -400,22 +471,84 @@ export class LunaSecStackDockerCompose {
...
-      TOKENIZER_URL: this.stackConfigOptions.tokenizerUrl,
+      LUNASEC_STACK_ENV: this.env,
+      TOKENIZER_URL: 'http://tokenizer-backend:37766',
       REACT_APP_TOKENIZER_URL: this.stackConfigOptions.tokenizerUrl,
-      CDN_HOST: 'localhost:8000',
+      CDN_HOST: `${this.getSecureFrameHostname()}:8000`,
+      LOCALSTACK_URL: `http://${this.getLocalstackHostname()}:4566`,
+      LOCAL_HTTPS_PROXY: 'https://localstack-proxy:4568',
+      LOCALSTACK_HOSTNAME: 'localstack',
+
```
Comment (factoidforrest) [inline js/sdks/packages/cli/src/docker-compose/lunasec-stack.ts:546]: dude when did this config get so thiiiiccc lol, nbd though

**Review by factoidforrest — APPROVED** [review]
very nice, the sheer amount of config changes here is astounding.  Im surprised we needed so much but im glad things are more configurable now.  

In the future keeping this kind of config as standardized as humanly possible is probably going to help our sanity.

File: js/demo-apps/packages/react-front-end/cypress/plugins/index.js (hunk 1 above)
Comment (freeqaz) [reply js/demo-apps/packages/react-front-end/cypress/plugins/index.js:47]: This hurt. We spent a long time figuring out WTF was going on. Painful!

**Pull request merged**